src/api/spec/models/obs_factory/openqa_job_spec.rb

```diff
@@ -0,0 +1,165 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe ObsFactory::OpenqaJob do
+  describe '.openqa_base_url' do
+    it { expect(ObsFactory::OpenqaJob.openqa_base_url).to eq('http://openqa.opensuse.org') }
+
+    context 'with a configured base_url' do
+      before do
+        CONFIG['openqa_base_url'] = 'http://configured-url.com'
+      end
+
+      it { expect(ObsFactory::OpenqaJob.openqa_base_url).to eq('http://configured-url.com') }
+    end
+  end
+
+  describe '.openqa_links_url' do
+    it { expect(ObsFactory::OpenqaJob.openqa_links_url).to eq('https://openqa.opensuse.org') }
+
+    context 'with a configured links_url' do
+      before do
+        CONFIG['openqa_links_url'] = 'http://links-url.com'
+      end
+
+      it { expect(ObsFactory::OpenqaJob.openqa_links_url).to eq('http://links-url.com') }
+    end
+  end
+
+  describe '#result_or_state' do
+    subject { ObsFactory::OpenqaJob.new(result: 'some-result', state: 'some-state') }
+
+    it { expect(subject.result_or_state).to eq('some-result') }
+
+    context 'with a result value of "none"' do
+      subject { ObsFactory::OpenqaJob.new(result: 'none', state: 'some-state') }
+
+      it { expect(subject.result_or_state).to eq('some-state') }
+    end
+  end
+
+  describe '#failing_modules' do
+    let(:modules) do
+      %w[passed softfailed running none failed].map do |result|
+        { 'name' => "mod_#{result}", 'result' => result }
+      end
+    end
+
+    subject { ObsFactory::OpenqaJob.new(modules: modules) }
+
+    it { expect(subject.failing_modules).to contain_exactly('mod_failed') }
+  end
+
+  describe '.find_all_by' do
+    let(:openqa_base_url) { 'http://openqa.opensuse.org/api/v1/jobs' }
+    let(:default_filter) { { scope: 'current' } }
+    let(:job1) { { 'id' => '1', 'name' => 'job1', 'iso' => 'iso1', 'state' => 'state1', 'build' => 'build1' } }
+    let(:job2) { { 'id' => '2', 'name' => 'job2', 'iso' => 'iso2', 'state' => 'state2', 'build' => 'build2' } }
+    let(:job3) { { 'id' => '3', 'name' => 'job3', 'Yiso' => 'iso3', 'state' => 'state3', 'build' => 'build3' } }
+
+    subject { ObsFactory::OpenqaJob.find_all_by(args, opt) }
+
+    before do
+      stub_request(:get, "#{openqa_base_url}?#{filters.to_query}").and_return(body: json_response.to_json, status: 200)
+    end
+
+    context 'without filters' do
+      let(:args) { {} }
```

Comment:
what about defining this one level up, so that it is not needed to repeat `let(:args) { {} }` everywhere, but just redefine it when it has a different value?